pipeline-runner/R/qc-7-embed_and_cluster.R

```diff
@@ -1,65 +1,78 @@
+run_clustering <- function(scdata, config, ignore_ssl_cert) {
+  message("starting clusters")
+  clustering_method <- config$clusteringSettings$method
+  methodSettings <-
+    config$clusteringSettings$methodSettings[[clustering_method]]
+  message("Running clustering")
+  cellSets <-
+    runClusters(clustering_method, methodSettings$resolution, scdata)
+  message("formatting cellsets")
+
+  formated_cell_sets <-
+    format_cluster_cellsets(cellSets, clustering_method, scdata@misc$color_pool)
+  message("updating through api")
+
+  replace_cell_class_through_api(
+    formated_cell_sets,
+    config$api_url,
+    scdata@misc$experimentId,
+    clustering_method,
+    config$auth_JWT,
+    ignore_ssl_cert
+  )
+}
+
+
 #' run clustering
 #'
 #'
 #' @param scdata seurat object
 #' @param config list with clustering parameters
 #' @param sample_id character
-#' @param cells_id list of cell ids that passed all filters so far
+#' @param cells_id list of cell ids that passed all filters so far, not necessary for this step
 #' @param task_name character
 #'
 #' @return list
 #' @export
 #'
-embed_and_cluster <-
- function(scdata,
-           config,
-           sample_id,
-           cells_id,
-           task_name = "configureEmbedding",
-           ignore_ssl_cert = FALSE) {
-
-    clustering_method <- config$clusteringSettings$method
-    methodSettings <-
-      config$clusteringSettings$methodSettings[[clustering_method]]
-    cellSets <-
-      runClusters(clustering_method, methodSettings$resolution, scdata)
-
-    formated_cell_sets <-
-      format_cluster_cellsets(cellSets, clustering_method, scdata@misc$color_pool)
-
-    replace_cell_class_through_api(
-      formated_cell_sets,
+embed_and_cluster <- function(
+  scdata,
+  config,
+  sample_id,
+  cells_id,
+  task_name = "configureEmbedding",
+  ignore_ssl_cert = FALSE
+) {
+
+  if (config$clustering_should_run == TRUE) {
+    run_clustering(scdata, config, ignore_ssl_cert)
+  } else {
+    message("Skipping clustering")
+  }
+
+  # add cl metadata if any
+  if (!is.null(config$metadata_s3_path)) {
+    cl_metadata_cellsets <- make_cl_metadata_cellsets(scdata, config)
+
+    # remove previously uploaded cell level metadata (to allow user to replace the file)
```

Comment:
this comment does not really apply anymore. the deletion is done by the API

Reply:
Removed